src/docfx/lib/HtmlUtility.cs

```diff
@@ -395,34 +402,52 @@ private static bool IsLink(ref HtmlToken token, in HtmlAttribute attribute)
         private static int CountWordInText(ReadOnlySpan<char> text)
         {
             var total = 0;
-            var word = false;
+            var isEnglishWord = false;
 
             foreach (var ch in text)
             {
-                if (ch == ' ' || ch == '\t' || ch == '\n' || ch == '\r')
+                if (IsCJKChar(ch))
                 {
-                    if (word)
+                    total++;
+
+                    if (isEnglishWord)
                     {
-                        word = false;
+                        isEnglishWord = false;
                         total++;
                     }
                 }
-                else if (
-                    ch != '.' && ch != '?' && ch != '!' &&
-                    ch != ';' && ch != ':' && ch != ',' &&
-                    ch != '(' && ch != ')' && ch != '[' &&
-                    ch != ']')
+                else
                 {
-                    word = true;
+                    if (ch == ' ' || ch == '\t' || ch == '\n' || ch == '\r')
+                    {
+                        if (isEnglishWord)
+                        {
+                            isEnglishWord = false;
+                            total++;
+                        }
+                    }
+                    else if (
+                        ch != '.' && ch != '?' && ch != '!' &&
+                        ch != ';' && ch != ':' && ch != ',' &&
+                        ch != '(' && ch != ')' && ch != '[' &&
+                        ch != ']')
+                    {
+                        isEnglishWord = true;
+                    }
                 }
             }
 
-            if (word)
+            if (isEnglishWord)
             {
                 total++;
             }
 
             return total;
         }
+
+        private static bool IsCJKChar(char c)
```

Comment:
This part of code is very performance sensitive (we are processing a huge number of markdown files like <https://github.com/MicrosoftDocs/azure-docs> and every character goes through this function), prefer `<` and `>` over regex.

Reply:
I have no better way than this, CJK character is so many, regex is the most commonly used.